config/internal/apiserver/deployment.yaml.tmpl

```diff
@@ -18,6 +18,54 @@ spec:
         component: data-science-pipelines
     spec:
       containers:
+        {{ if .APIServer.EnableRoute }}
+        - name: oauth-proxy
+          args:
+            - --https-address=:8443
+            - --provider=openshift
+            - --openshift-service-account=ds-pipeline-{{.Name}}
+            - --upstream=http://localhost:8888
+            - --tls-cert=/etc/tls/private/tls.crt
+            - --tls-key=/etc/tls/private/tls.key
+            - --cookie-secret=SECRET
+            - '--openshift-delegate-urls={"/": {"group":"route.openshift.io","resource":"routes","verb":"get","name":"ds-pipeline-{{.Name}}","namespace":"{{.Namespace}}"}}'
+            - '--openshift-sar={"namespace":"{{.Namespace}}","resource":"routes","resourceName":"ds-pipeline-{{.Name}}","verb":"get","resourceAPIGroup":"route.openshift.io"}'
```

Comment:
I'm not understanding why we need both `delegate-urls` and `sar`. From my brief reading of [this doc](https://github.com/openshift/oauth-proxy#limiting-access-to-users), it sounds like they are competing mechanisms. Maybe I have that wrong.

Reply:
openshift-delegate-urls >> for  token  based authentication 
openshift-sar >> console/ui based authentication 

@harshad16 can expand more when he's back, as he's recently done deep looking into oauth proxy